src/main/java/com/smalaca/messagesender/service/MessageSenderService.java

```diff
@@ -0,0 +1,35 @@
+package com.smalaca.messagesender.service;
+
+import com.smalaca.messagesender.domain.Message;
+import com.smalaca.messagesender.domain.MessageRepository;
+import com.smalaca.messagesender.exceptions.inmemory.NoMessageException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MessageSenderService {
+
+    private MessageRepository messageRepository;
+    private FakeMailSender fakeMailSender;
+
+    @Autowired
+    public MessageSenderService(MessageRepository messageRepository, FakeMailSender fakeMailSender) {
+        this.messageRepository = messageRepository;
+        this.fakeMailSender = fakeMailSender;
+    }
+
+    public Response sendMessageViaEmail(String id) throws NoMessageException {
+
+        try {
```

Comment:
w idealnym kodzie try-catch powinien wylecieć. Ze względu na fakt, że to są ćwiczenia i chcesz wyrzucić ten exception - usuń if'a